src/crimson/os/seastore/cache.cc

```diff
@@ -501,6 +501,52 @@ void Cache::register_metrics()
       );
     }
   };
+  auto register_lba_tree_metrics = [this](
+      const sm::label_instance& tree_label,
+      lba_tree_efforts_t& committed_lba_tree_efforts,
+      lba_tree_efforts_t& invalidated_lba_tree_efforts) {
+    metrics.add_group(
+      "cache",
+      {
+        sm::make_counter(
+          "lba_tree_alloc_extent_committed",
+          committed_lba_tree_efforts.num_alloc_extents,
+          sm::description("total number of committed lba alloc_extent operations"),
+          {tree_label}
+        ),
+        sm::make_counter(
+          "lba_tree_iter_prev_committed",
+          committed_lba_tree_efforts.num_iter_prevs,
+          sm::description("total number of committed lba iter_prev operations"),
+          {tree_label}
+        ),
+        sm::make_counter(
+          "lba_tree_iter_next_committed",
+          committed_lba_tree_efforts.num_iter_nexts,
+          sm::description("total number of committed lba iter_next operations"),
+          {tree_label}
+        ),
+        sm::make_counter(
+          "lba_tree_alloc_extent_invalidated",
+          invalidated_lba_tree_efforts.num_alloc_extents,
+          sm::description("total number of invalidated lba alloc_extent operations"),
+          {tree_label}
+        ),
+        sm::make_counter(
+          "lba_tree_iter_prev_invalidated",
+          invalidated_lba_tree_efforts.num_iter_prevs,
+          sm::description("total number of invalidated lba iter_prev operations"),
+          {tree_label}
+        ),
+        sm::make_counter(
+          "lba_tree_iter_next_invalidated",
+          invalidated_lba_tree_efforts.num_iter_nexts,
+          sm::description("total number of invalidated lba iter_next operations"),
+          {tree_label}
+        ),
+      }
+    );
+  };
```

Comment:
I think the goal is to measure the number of iterations vs the number of allocations during laddr-alloc, but I don't think it worths to relate to transaction invalidate/commit in `Cache`.

So could you define these metrics inside the lba tree instead?

Reply:
sure, will do it.